### DISCUSSION TITLE
is there a way to choose a controlnet pose from a folder of poses - 1 at a time at random - instead of run the whole batch?

### DISCUSSION THREAD

**Discussion by Sacred-Reputation** [vladmandic/sdnext#1853] [General]
i have a folder of lineart images - i'd like it to choose 1 at random each time i hit generate. not sure how to setup controlnet or dynamic prompts or X/Y/Z grid - somehow have it call a different lineart image out of the folder for each generation.

**Comment by brknsoul** [comment]
Yes... somewhat ;-)
Make a folder somewhere (eg `automatic\models\ControlNet\test`) and place your lineart images in there.
Then enable ControlNet, switch to Batch mode, insert the created folder,

Once all your settings are as you like them, hit "Generate". This will create 1 image for each lineart image you have in that folder.

![image](https://github.com/vladmandic/automatic/assets/1969381/7055cc43-8c87-4122-a926-89ac4d3446bc)

(this example is for OpenPose, but Lineart should function similarly).

**Reply by Sacred-Reputation** [reply]
Right - i've got that. What i'm trying to achieve is for it to randomly select from ONE of the poses in that folder for each generation, NOT run the whole batch. Any ideas?
